checkov/common/images/image_referencer.py

```diff
@@ -75,3 +88,149 @@ def inspect(image_name: str) -> str:
+        root_path = Path(root_path) if root_path else None
+        check_class = f"{image_scanner.__module__}.{image_scanner.__class__.__qualname__}"
+        report_type = CheckType.SCA_IMAGE
+
+        for image in images:
+            self.add_image_record(
+                report=report,
+                root_path=root_path,
+                check_class=check_class,
+                dockerfile_path=image.file_path,
+                image=image,
+                runner_filter=runner_filter,
+                report_type=report_type,
+            )
+
+        return report
+
+    def add_image_record(
+        self,
+        report: Report,
+        root_path: Path | None,
+        check_class: str,
+        dockerfile_path: str,
+        image: Image,
+        runner_filter: RunnerFilter,
+        report_type: str,
+    ) -> None:
+        """Adds an image record to the given report, if possible"""
+
+        cached_results: dict[str, Any] = image_scanner.get_scan_results_from_cache(f"image:{image.name}")
+        if cached_results:
+            logging.info(f"Found cached scan results of image {image.name}")
+
+            result = cached_results.get("results", [{}])[0]
+            vulnerabilities = result.get("vulnerabilities") or []
+            image_id = self.extract_image_short_id(result)
+            image_details = self.get_image_details_from_twistcli_result(scan_result=result, image_id=image_id)
+            if root_path:
+                try:
+                    dockerfile_path = str(Path(dockerfile_path).relative_to(root_path))
+                except ValueError:
+                    # Path.is_relative_to() was implemented in Python 3.9
+                    pass
+            rootless_file_path = dockerfile_path.replace(Path(dockerfile_path).anchor, "", 1)
+
+            parse_vulns_to_records(
+                report=report,
+                check_class=check_class,
+                scanned_file_path=os.path.abspath(dockerfile_path),
+                rootless_file_path=f"{rootless_file_path} ({image.name} lines:{image.start_line}-{image.end_line} ({image_id}))",
+                runner_filter=runner_filter,
+                vulnerabilities=vulnerabilities,
+                packages=[],
+                license_statuses=[],
+                image_details=image_details,
+                report_type=report_type,
+            )
+        elif strtobool(os.getenv("CHECKOV_EXPERIMENTAL_IMAGE_REFERENCING", "False")):
+            # experimental flag on running image referencers via local twistcli
+            from checkov.sca_image.runner import Runner as sca_image_runner
+
+            runner = sca_image_runner()
+
+            image_id = ImageReferencer.inspect(image.name)
+            scan_result = runner.scan(image_id, dockerfile_path, runner_filter)
+            if scan_result is None:
+                return None
+
+            self.raw_report = scan_result
+            result = scan_result.get('results', [{}])[0]
```

Comment:
you use the next logic twice. please make sure to reduce it as much as possible in your next PR